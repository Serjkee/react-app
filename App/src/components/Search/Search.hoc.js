import {connect} from 'react-redux'
import {changingSearchInputValue} from '../../redux/filters/filters.actions'
import {asyncHandlingSubmitFormAction} from '../../redux/movies/movies.actions'
import Search from './Search'


const mapStateToProps = (state) => ({searchValue: state.filters.searchValue, searchBy: state.filters.searchBy, sortBy: state.filters.sortBy})
const mapDispatchToProps = (dispatch) => ({
  changingInputValue(e) {
    dispatch(changingSearchInputValue(e.target.value))
  },
  onSubmitForm(searchText, searchBy, sortBy) {
    dispatch(asyncHandlingSubmitFormAction(searchText, searchBy, sortBy));
  }
})

export default  connect(mapStateToProps, mapDispatchToProps)(Search)