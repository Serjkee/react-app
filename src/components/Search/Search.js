import React, { Component } from 'react'

import SearchInput from './SearchInput/SearchInput'
import SearchButton from './SearchButton/SearchButton'
import SearchFilters from './SearchFilters/SearchFilters'
import './Search.scss'

export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {clicked: ''}
  }

  submitForm(e) {
    e.preventDefault();
    this.setState({clicked: 'event fired'});
  }

  render() {
    return (
      <form onSubmit={this.submitForm.bind(this)}>
        <SearchInput placeholder='Example: Quentin Tarantino' inputClass='search' />
        <div className='header-filters' >
          <SearchFilters />
          <SearchButton buttonText='Search' buttonClass='filter' />
        </div>
      </form>
    )
  }
}

export default Search
