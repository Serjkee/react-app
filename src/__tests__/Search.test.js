import React from 'react'
import Search from '../components/Search/Search'

describe('Tests Search component', () => {
  it('Matching snapshot', () => {
    const component = shallow(<Search />)

    expect(component).toMatchSnapshot();
  })

  it('Checking submit event', () => {
    const component = mount(<Search />)

    component.find('form').simulate('submit')
    expect(component.state().clicked).toEqual('event fired')
  })
})
