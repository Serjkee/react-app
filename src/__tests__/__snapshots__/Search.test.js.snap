// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Search component Matching snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <SearchInput
    inputClass="search"
    placeholder="Example: Quentin Tarantino"
  />
  <div
    className="header-filters"
  >
    <SearchFiltes />
    <SearchButton
      buttonClass="filter"
      buttonText="Search"
    />
  </div>
</form>
`;
